src/OnboardingSPA/steps/SiteGen/SiteDetails/walkthrough/index.js

```diff
@@ -0,0 +1,259 @@
+		};
+	} );
+
+	const { setCurrentOnboardingData } = useDispatch( nfdOnboardingStore );
+
+	useEffect( () => {
+		if ( currentData.sitegen.siteDetails?.name !== '' ) {
+			setCustomerInputName( currentData.sitegen.siteDetails.name );
+			setEditing( true );
+		}
+		if ( currentData.sitegen.siteDetails?.type !== '' ) {
+			setCustomerInputType( currentData.sitegen.siteDetails.type );
+		}
+		if ( currentData.sitegen.siteDetails?.style !== '' ) {
+			setCustomerInputStyle( currentData.sitegen.siteDetails.style );
+		}
+		if ( currentData.sitegen.siteDetails?.uniqueAboutBusiness !== '' ) {
+			setCustomerInputUnique(
+				currentData.sitegen.siteDetails.uniqueAboutBusiness
+			);
+		}
+		getSiteDetails();
+	}, [] );
+
+	async function getSiteDetails() {
+		let siteDetailsmetas = await getSiteDetailsmeta();
+		siteDetailsmetas = siteDetailsmetas.body;
+		siteDetailsmetas = JSON.parse( siteDetailsmetas ).reduce(
+			( acc, item ) => {
+				const { field, question, prompt, placeholder } = item;
+				acc[ `${ field }` ] = question;
+				acc[ `${ field }PromptText` ] = prompt;
+				acc[ `${ field }Placeholder` ] = placeholder;
+
+				return acc;
+			},
+			{}
+		);
+		setSiteDetailsmeta( siteDetailsmetas );
+	}
+
+	const handlePromptChange = ( field, e ) => {
+		e.preventDefault();
+		currentData.sitegen.siteDetails[ field ] = e.target.value;
+		setCurrentOnboardingData( currentData );
+
+		switch ( field ) {
+			case 'name':
+				setCustomerInputName( e.target.value );
+				break;
+			case 'type':
+				setCustomerInputType( e.target.value );
+				break;
+			case 'style':
+				setCustomerInputStyle( e.target.value );
+				break;
+			case 'uniqueAboutBusiness':
+				setCustomerInputUnique( e.target.value );
+				break;
+			default:
+				break;
+		}
+	};
+
+	const concatenatePrompt = () => {
+		const siteDetails = currentData.sitegen.siteDetails;
+		let concatenatedString = '';
+
+		for ( const field in siteDetails ) {
+			if ( siteDetails[ field ] ) {
+				switch ( field ) {
+					case 'name':
+						concatenatedString += `${ siteDetailsmeta.businessNamePromptText } ${ siteDetails[ field ] }.`;
+						break;
+					case 'type':
+						concatenatedString += `${ siteDetailsmeta.websiteTypePromptText } ${ siteDetails[ field ] }.`;
+						break;
+					case 'style':
+						concatenatedString += `${ siteDetailsmeta.writeStylePromptText } "${ siteDetails[ field ] }".`;
+						break;
+					case 'uniqueAboutBusiness':
+						concatenatedString += `${ siteDetailsmeta.uniqueBusinessPromptText } ${ siteDetails[ field ] }.`;
+						break;
+					default:
+						break;
+				}
+			}
+		}
+		return concatenatedString;
+	};
+
+	const selectOption = ( event, classname ) => {
+		document.querySelectorAll( `.${ classname }` ).forEach( ( button ) => {
+			button.classList.remove(
+				'nfd-sg-site-details-rows-button-selected'
+			);
+		} );
+		event.target.classList.add(
+			'nfd-sg-site-details-rows-button-selected'
+		);
```

Comment:
Let's change the classes with React code using `classnames` and not Directly manipulate the DOM.